pkgs/by-name/mi/mirrord/package.nix

```
@@ -0,0 +1,55 @@
+{
+  lib,
+  stdenv,
+  fetchurl,
+  testers,
+  mirrord,
+  autoPatchelfHook,
+}:
+
+let
+  manifest = lib.importJSON ./manifest.json;
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "mirrord";
+  version = manifest.version;
+
+  src =
+    let
+      system = stdenv.hostPlatform.system;
+    in
+    fetchurl (manifest.assets.${system} or (throw "Unsupported system: ${system}"));
```

It won't evaluate for platforms that are not listed in `meta.platforms` anyway.
```suggestion
    fetchurl (manifest.assets.${system});
```